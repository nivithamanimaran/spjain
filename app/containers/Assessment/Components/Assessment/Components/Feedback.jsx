import React, { useState } from 'react';
import { Box, Paper, Typography } from '@material-ui/core';
import KenSlider from '../../../../../components/KenSlider';
import KenInputField from '../../../../../components/KenInputField/index';
import { useTranslation } from 'react-i18next';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
import { makeStyles } from '@material-ui/core/styles';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import Hidden from '@material-ui/core/Hidden';

const useStyles = makeStyles(theme => ({
  paper: {
    marginBottom: '10px',
  },
  paperBorder: {
    border: `0.5px solid ${theme.palette.KenColors.neutral40}`,
  },
  btnBox: {
    display: 'flex',
    cursor: 'pointer',
    width: 'fit-content',
    alignItems: 'center',
  },
  btnLabel: {
    marginLeft: '5px',
    fontWeight: '600',
    fontSize: '14px',
  },
  container: {
    // display: 'flex',
    alignItems: 'center',
    width: '100%',
  },
  resetBtn: {
    [theme.breakpoints.up('md')]: {
      marginLeft: '10px',
    },
    [theme.breakpoints.down('md')]: {
      marginLeft: '0px',
    },
  },
  slider: {
    minHeight: '400px',
  },
}));

export default function Feedback(props) {
  const { data } = props;
  const { t } = useTranslation();
  const classes = useStyles();
  const [feedbacks, setFeedbacks] = useState(data || []);

  const handleAddFeedback = () => {
    const fBacks = [...feedbacks];
    fBacks.push({
      range: [100, 50],
      feedback: '',
    });
    setFeedbacks([...fBacks]);
  };
  const handleRemoveFeedback = index => {
    const fBacks = [...feedbacks];
    if (index > -1) {
      fBacks.splice(index, 1);
    }
    setFeedbacks([...fBacks]);
  };

  const handleResetFeedback = () => {
    setFeedbacks([]);
  };

  const marks = [
    {
      value: 100,
      label: '100%',
    },
    {
      value: 90,
      label: '90%',
    },
    {
      value: 80,
      label: '80%',
    },
    {
      value: 70,
      label: '70%',
    },
    {
      value: 60,
      label: '60%',
    },
    {
      value: 50,
      label: '50%',
    },
    {
      value: 40,
      label: '40%',
    },
    {
      value: 30,
      label: '30%',
    },
    {
      value: 20,
      label: '20%',
    },
    {
      value: 10,
      label: '10%',
    },
    {
      value: 0,
      label: '0%',
    },
  ];

  function valueText(value) {
    return `${value}%`;
  }

  const handleRangeChange = (event, newValue, index) => {
    let fbs = [...feedbacks];
    fbs[index].range = newValue;
    setFeedbacks(fbs);
  };

  const handleFeedbackChange = (event, index) => {
    let fbs = [...feedbacks];
    fbs[index].feedback = event.target.value;
    setFeedbacks(fbs);
  };

  const Button = btnProps => {
    const { handleClick, label, startIcon, className, dataTestid } = btnProps;
    return (
      <Box
        onClick={handleClick}
        className={`${classes.btnBox} ${className}`}
        data-testid={dataTestid || 'button'}
      >
        {startIcon}
        <Typography
          className={classes.btnLabel}
          component="span"
          color="primary"
        >
          {label}
        </Typography>
      </Box>
    );
  };

  return (
    <>
      <Box mb={3}>
        <Paper
          elevation={0}
          className={
            feedbacks?.length > 0
              ? `${classes.paper} ${classes.paperBorder}`
              : classes.paper
          }
          data-testid="feedback-list"
        >
          {feedbacks?.map((f, index) => {
            return (
              <>
                <Box p={2} data-testid="feedback-list-item">
                  <Box mb={3}>
                    <Button
                      dataTestid="remove"
                      handleClick={() => handleRemoveFeedback(index)}
                      //   label={`Remove feedback ${index + 1} : ${f.feedback}`}
                      label={`Remove feedback`}
                      startIcon={
                        <RemoveCircleOutlineIcon
                          fontSize="small"
                          color="primary"
                        />
                      }
                    />
                  </Box>
                  <Box mb={3}>
                    <Hidden smUp>
                      <KenSlider
                        orientation="vertical"
                        minHeight="400px"
                        valueText={valueText}
                        value={f.range}
                        handleChange={(e, n) => handleRangeChange(e, n, index)}
                        marks={marks}
                        label={`Grade ${index + 1}`}
                        track={true}
                        className={classes.slider}
                        min={10}
                        max={50}
                      />
                    </Hidden>
                    <Hidden smDown>
                      <KenSlider
                        valueText={valueText}
                        value={f.range}
                        handleChange={(e, n) => handleRangeChange(e, n, index)}
                        marks={marks}
                        label={`Grade ${index + 1}`}
                        track={true}
                      />
                    </Hidden>
                  </Box>
                  <Box mb={3} data-testid="feedback-input">
                    <KenInputField
                      label={`Feedback ${index + 1}`}
                      value={f.feedback}
                      onChange={e => handleFeedbackChange(e, index)}
                    />
                  </Box>
                </Box>
              </>
            );
          })}
        </Paper>

        <Box
          display={{ xs: 'block', md: 'flex' }}
          className={classes.container}
        >
          <Button
            dataTestid="add"
            handleClick={handleAddFeedback}
            label={
              feedbacks?.length > 0
                ? 'Add more range for feedback'
                : 'Add overall feedback'
            }
            startIcon={
              <AddCircleOutlineIcon fontSize="small" color="primary" />
            }
          />
          {feedbacks?.length > 0 && (
            <Button
              dataTestid="reset"
              handleClick={handleResetFeedback}
              label="Reset"
              startIcon={<HighlightOffIcon fontSize="small" color="primary" />}
              className={classes.resetBtn}
            />
          )}
        </Box>
      </Box>
    </>
  );
}
