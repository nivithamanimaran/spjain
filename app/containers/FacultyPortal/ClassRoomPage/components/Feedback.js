import React from 'react'
import Rating from '@material-ui/lab/Rating';
import { Grid, Box, Typography, Chip, Paper, makeStyles, Divider } from '@material-ui/core';
import CircularProgressWithLabel from '../../../../components/ChartWidgets/CircularProgressWithLabel';
import Like from '../../../../assets/icons/Like.svg';
import DisLike from '../../../../assets/icons/Dislike.svg';
import { useTranslation } from 'react-i18next';




const feeds = [
    {
        comment: "The wording of these reviews is a bit misleading. He is a great teacher and makes the material interesting, but this class is NOT an easy A. He graded easier than my last professor, but I would work hard on my papers and get Cs. But he always explained the grade and helped me improve in meetings. Took a lot of time to help me. I ended with a B. ",
        tags: [
            "Strict", "Hardworking"
        ],
        date: "8, Jul, 2020",
        rating: 3.5,
        likes: 20,
        dislikes: 0
    },
    {
        comment: "Thanks for taking the load while I travel around - you're the best. Always in a good mood even when more tasks are heaped on you!",
        tags: [
            "Caring", "Hardworking"
        ],
        date: "10, Sep, 2020",
        rating: 3,
        likes: 10,
        dislikes: 20
    },

]

const feedOverall = {
    rating: 3.5,
    reviews: 26,
    percentage: 50,
    tags: [
        "Strict",
        "Caring",
        "Hard Working"
    ]
}

const useStyles = makeStyles(theme => ({
    cardHeader:{
        fontSize:'16px',
        color:theme.palette.KenColors.neutral100
    },
    rating:{
        fontSize:'48px',
        color:theme.palette.KenColors.primary
    },
    chip: {
        marginRight: theme.spacing(1),
        fontSize:'12px',
        color: theme.palette.KenColors.neutral100,
        background:theme.palette.KenColors.neutral20,
    },
    feedLike: {
        height: '1rem'
    },
    feedLikeText: {
        fontSize: '0.725rem',
        padding: 3
    },
    feedItemGrid: {
        marginTop: theme.spacing(2),
        fontSize:'14px',
        color: theme.palette.KenColors.neutral400,
    },
    textFont:{
        fontSize:'14px',
        color: theme.palette.KenColors.neutral400,
    },
}))

function Feedback(props) {
    const {t} = useTranslation();
    const classes = useStyles();
    return (

        <Paper>
            <Box p={2}>
                <Typography className={classes.cardHeader}>
                    {t("headings:Feedback")}
                </Typography>
                {/* over all section */}
                <Box py={2}>
                    <Grid container spacing={3}>

                        <Grid item >
                            <Typography variant="h4" className={classes.rating}>
                                {feedOverall.rating}
                            </Typography>
                        </Grid>
                        <Grid item >
                            <Rating name="overall-rating" defaultValue={feedOverall.rating}
                                 precision={0.5} />
                            
                            <Typography className={classes.textFont}>

                                {feedOverall.reviews} {t("Reviews")}
                            </Typography>
                        </Grid>
                        <Grid item>
                            <Grid container spacing={1}>
                                <Grid item>
                                <CircularProgressWithLabel value={feedOverall.percentage} label={feedOverall.percentage + "%"} />
                                </Grid>
                                <Grid item sm={6} md={6} lg={6}> 
                                <Typography className={classes.textFont}>
                                {t("Would_Attend_Again")}
                            </Typography>
                                </Grid>
                            </Grid>
                        </Grid>

                        <Grid item>
                            <Box mb={1}>
                            <Typography className={classes.textFont}>
                                {t("Top_Tags")}
                            </Typography>
                            </Box>
                            {feedOverall.tags && feedOverall.tags.map(tag => <Chip variant="outlined" label={tag} size="small" className={classes.chip} />)}

                        </Grid>
                    </Grid>
                </Box>
                <Divider />

                {/* Feedbacks */}
                {feeds && feeds.map(feed => <FeedbackItem data={feed} />)}
            </Box>
        </Paper>
    )
}

Feedback.propTypes = {

}

function FeedbackItem({ data }) {
    const classes = useStyles();
    return <Box py={2}>
        <Grid container justify="space-between" >
            <Grid item >
                <Rating name="rating" defaultValue={data.rating} precision={0.5} size="small" />
            </Grid>
            <Grid item>
                <Typography className={classes.textFont}>{data.date}</Typography>
            </Grid>
        </Grid>
        <Typography className={classes.feedItemGrid}>
            {data.comment}
        </Typography>

        <Grid container justify="space-between" className={classes.feedItemGrid}>
            <Grid item >
                {data.tags && data.tags.map(tag => <Chip variant="outlined" size="small" label={tag} className={classes.chip} />)}
            </Grid>
            <Grid item>
                <img src={Like} alt="like" className={classes.feedLike} />
                <Typography className={classes.feedLikeText} display="inline">
                    {data.likes}
                </Typography>
                &nbsp;&nbsp;
                <img src={DisLike} alt="dis-like" className={classes.feedLike} />
                <Typography className={classes.feedLikeText} display="inline">
                    {data.dislikes}
                </Typography>
            </Grid>
        </Grid>

    </Box>
}

export default Feedback

