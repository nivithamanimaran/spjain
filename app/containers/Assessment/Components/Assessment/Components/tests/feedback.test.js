import React from 'react';
import {
  render,
  fireEvent,
  within,
  queryByAttribute,
} from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { screen } from '@testing-library/dom';

import Feedback from '../Feedback';

const feedbacks = [
  { range: [75, 100], feedback: 'Good' },
  { range: [50, 74], feedback: 'Average' },
];

it('Feedback should render and match the snapshot without props', () => {
  const {
    container: { firstChild },
  } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(firstChild).toMatchSnapshot();
});

it('Feedback should render correctly with props', () => {
  const {
    container: { firstChild },
  } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(firstChild).toMatchSnapshot();
});

it('should render list of 2 feedbacks', () => {
  render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );

  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );
  const list = queryByTestId('feedback-list', {
    name: /feedbacks/i,
  });
  const { queryAllByTestId } = within(list);
  const items = queryAllByTestId('feedback-list-item');
  expect(items.length).toBe(2);
});

it('should update state on feedback change', () => {
  const setFeedback = jest.fn();

  const handleClick = jest.spyOn(React, 'useState');
  handleClick.mockImplementation(feedback => [feedback, setFeedback]);

  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback />
      </ThemeProvider>
    </IntlProvider>
  );

  const node = queryByTestId('add');
  fireEvent.click(node);
  expect(setFeedback).toBeTruthy();
});

it('should add feedback', () => {
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );

  const list = queryByTestId('feedback-list', {
    name: /feedbacks/i,
  });
  const { queryAllByTestId } = within(list);
  const items = queryAllByTestId('feedback-list-item');
  const initLength = items.length;
  expect(items.length).toBe(items.length);

  const node = queryByTestId('add');
  fireEvent.click(node);
  const updatedItems = queryAllByTestId('feedback-list-item');

  expect(updatedItems.length).toBe(initLength + 1);
});

it('Feedback input fields, remove and reset buttons should render correctly after addition of feedback', () => {
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );

  const node = queryByTestId('add');
  const removeNode = queryByTestId('remove');
  const resetNode = queryByTestId('reset');
  fireEvent.click(node);
  const inputNode = queryByTestId('feedback-input');
  expect(removeNode).toBeTruthy();
  expect(resetNode).toBeTruthy();
  expect(inputNode).toBeTruthy();
});

it('should remove feedback', () => {
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );

  const list = queryByTestId('feedback-list', {
    name: /feedbacks/i,
  });
  const { queryAllByTestId } = within(list);
  const items = queryAllByTestId('feedback-list-item');
  const initLength = items.length;
  expect(items.length).toBe(items.length);

  const node = queryByTestId('remove');
  fireEvent.click(node);
  const updatedItems = queryAllByTestId('feedback-list-item');

  expect(updatedItems.length).toBe(initLength - 1);
});

it('should reset feedback and disappear reset and remove buttons', () => {
  const { queryByTestId } = render(
    <IntlProvider locale="en">
      <ThemeProvider>
        <Feedback data={feedbacks} />
      </ThemeProvider>
    </IntlProvider>
  );

  const list = queryByTestId('feedback-list', {
    name: /feedbacks/i,
  });
  const { queryAllByTestId } = within(list);
  const items = queryAllByTestId('feedback-list-item');
  expect(items.length).toBe(items.length);

  const node = queryByTestId('reset');
  fireEvent.click(node);
  const updatedItems = queryAllByTestId('feedback-list-item');
  const removeNode = queryByTestId('remove');
  const resetNode = queryByTestId('reset');

  expect(updatedItems.length).toBe(0);
  expect(removeNode).toBeNull();
  expect(resetNode).toBeNull();
});
