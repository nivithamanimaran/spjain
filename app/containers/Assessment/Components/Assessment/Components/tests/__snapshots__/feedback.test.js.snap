// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should render and match the snapshot without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-8"
>
  <div
    class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation0 MuiPaper-rounded"
    data-testid="feedback-list"
  />
  <div
    class="MuiBox-root MuiBox-root-9 makeStyles-container-5"
  >
    <div
      class="MuiBox-root MuiBox-root-10 makeStyles-btnBox-3 undefined"
      data-testid="add"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-btnLabel-4 MuiTypography-body1 MuiTypography-colorPrimary"
      >
        Add overall feedback
      </span>
    </div>
  </div>
</div>
`;

exports[`Feedback should render correctly with props 1`] = `
<div
  class="MuiBox-root MuiBox-root-18"
>
  <div
    class="MuiPaper-root makeStyles-paper-11 makeStyles-paperBorder-12 MuiPaper-elevation0 MuiPaper-rounded"
    data-testid="feedback-list"
  >
    <div
      class="MuiBox-root MuiBox-root-19"
      data-testid="feedback-list-item"
    >
      <div
        class="MuiBox-root MuiBox-root-20"
      >
        <div
          class="MuiBox-root MuiBox-root-21 makeStyles-btnBox-13 undefined"
          data-testid="remove"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <span
            class="MuiTypography-root makeStyles-btnLabel-14 MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Remove feedback
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-22"
      />
      <div
        class="MuiBox-root MuiBox-root-23"
        data-testid="feedback-input"
      >
        <div
          class="MuiFormControl-root makeStyles-formControl-25"
        >
          <div
            class="makeStyles-labelWrap-34"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-label-32 MuiTypography-body1"
                >
                  
                  Feedback 1
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-optional-33 MuiTypography-body1"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-36 makeStyles-inputBorder-31 MuiInputBase-formControl MuiInputBase-multiline"
            pattern="[A-Za-z]{3}"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-38 MuiInputBase-inputMultiline"
              rows="1"
              style="height: 0px; overflow: hidden;"
            >
              Good
            </textarea>
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-38 MuiInputBase-inputMultiline"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
              tabindex="-1"
            />
          </div>
          <p
            class="MuiTypography-root makeStyles-typoSupporting-26 MuiTypography-body1 MuiTypography-alignLeft"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-39"
      data-testid="feedback-list-item"
    >
      <div
        class="MuiBox-root MuiBox-root-40"
      >
        <div
          class="MuiBox-root MuiBox-root-41 makeStyles-btnBox-13 undefined"
          data-testid="remove"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <span
            class="MuiTypography-root makeStyles-btnLabel-14 MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Remove feedback
          </span>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-42"
      />
      <div
        class="MuiBox-root MuiBox-root-43"
        data-testid="feedback-input"
      >
        <div
          class="MuiFormControl-root makeStyles-formControl-25"
        >
          <div
            class="makeStyles-labelWrap-34"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-label-32 MuiTypography-body1"
                >
                  
                  Feedback 2
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <p
                  class="MuiTypography-root makeStyles-optional-33 MuiTypography-body1"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-36 makeStyles-inputBorder-31 MuiInputBase-formControl MuiInputBase-multiline"
            pattern="[A-Za-z]{3}"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-38 MuiInputBase-inputMultiline"
              rows="1"
              style="height: 0px; overflow: hidden;"
            >
              Average
            </textarea>
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-38 MuiInputBase-inputMultiline"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
              tabindex="-1"
            />
          </div>
          <p
            class="MuiTypography-root makeStyles-typoSupporting-26 MuiTypography-body1 MuiTypography-alignLeft"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-44 makeStyles-container-15"
  >
    <div
      class="MuiBox-root MuiBox-root-45 makeStyles-btnBox-13 undefined"
      data-testid="add"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-btnLabel-14 MuiTypography-body1 MuiTypography-colorPrimary"
      >
        Add more range for feedback
      </span>
    </div>
    <div
      class="MuiBox-root MuiBox-root-46 makeStyles-btnBox-13 makeStyles-resetBtn-16"
      data-testid="reset"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-btnLabel-14 MuiTypography-body1 MuiTypography-colorPrimary"
      >
        Reset
      </span>
    </div>
  </div>
</div>
`;
